src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -275,6 +284,29 @@ private void addOrderByToSelect(Select select, MySqlSelectQueryBlock queryBlock,
         }
     }
 
+    private SQLExpr extractExprFromOrderExpr(SQLSelectOrderByItem sqlSelectOrderByItem) {
+        SQLExpr expr = sqlSelectOrderByItem.getExpr();
+        if (hasNullOrderInBinaryOrderExpr(expr)) {
```

Comment:
Could you put comment here why this is necessary and how it works?
I don't see it in the PR description and have hard time understanding what's going on.

Reply:
Sure, will do.

Reply:
Added clarifying comment.